File: frontend/src/types/payment.ts
import type { PaginationResponse } from "./pagination";

export interface Payment {
  id: string;
  merchant_id: number;
  merchant_name: string;
  amount: string;
  status: "completed" | "processing" | "failed";
  created_at: string;
}

export interface PaymentListResponse extends PaginationResponse<Payment> {
  payments: Payment[];
}

File: backend/internal/entity/payment.go
package entity

import "time"

type Payment struct {
	ID           string    `json:"id"`
	MerchantID   int       `json:"merchant_id"`
	MerchantName string    `json:"merchant_name,omitempty"`
	Amount       string    `json:"amount"`
	Status       string    `json:"status"`
	CreatedAt    time.Time `json:"created_at"`
}
